astacus/coordinator/list.py

```diff
@@ -3,12 +3,48 @@
 See LICENSE for details
 
 """
-
 from astacus.common import ipc, magic
-from astacus.common.storage import MultiStorage
+from astacus.common.storage import JsonStorage, MultiStorage
+from collections import defaultdict
+from collections.abc import Iterator
+
+
+def compute_deduplicated_snapshot_file_stats(manifest: ipc.BackupManifest) -> tuple[int, int]:
+    """Compute stats over snapshot files as identified by their hex digest.
+
+    There may be duplicate hex digests within nodes for multiple copies of the same data chunks.
+    Duplicates' size are aggregated as they require physical disk space during restore.
+    On the other hand, hex digests can be safely deduplicated across nodes. The ocurrence with
+    the highest associated files' count and maximum file size across all nodes is kept.
+    """
+    global_file_counter = defaultdict(list)
+    global_file_sizes = defaultdict(list)
+    num_files = 0
+    total_size = 0
+    for snapshot_result in manifest.snapshot_results:
+        if not snapshot_result.state:
+            num_files += snapshot_result.files
+            total_size += snapshot_result.total_size
```

Comment:
This is probably not necessary. `state` is optional in typing so that empty result can be initialized, but if it is not set, these values are also not set (see node/snapshot.py) and the backup has failed so manifest is unlikely to exist. Could just skip these (+- asserting state is set) and calculate `num_files` and `total_size` in the end without the initial `= 0` and the `+=`s here.

(The intermediate result without it set is only used for progress reporting and while decoupling these two would be possible, I haven't bothered to.)

This would be perhaps worth documenting in ipc.py too.

Reply:
Removed the dead code and added a comment in `ipc`